### PR TITLE
fix(lists): entry list items are now full clickable to select.

### LINKED ISSUE #1764: Remove link type from iteration panel and work item names
In the iteration panel, iteration names should not be blue links and the entire iteration row should be selectable.  Also captured in this issue:  https://github.com/fabric8io/fabric8-planner/issues/1717

Also for individual work items in the list view and board view, the names should not be links, but rather just black text.  Selecting the whole row or the whole card should bring up the work item detail panel.

This PR caused this change in behavior: https://github.com/fabric8io/fabric8-planner/pull/1688

### PR DESCRIPTION
**Pull request by michaelkleinhenz**
Fixes #1764 and #1717

### CODE REVIEW

**Comment by fabric8cd** [comment]
@michaelkleinhenz snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1766-fabric8-planner.int.rdu2c.fabric8.io

File: src/app/iteration/iteration.component.html [hunk 1]
```diff
@@ -11,17 +11,16 @@
         </h3>
       </div>
       <ul class="iteration-expanded-div" [collapse]="isCollapsedCurrentIteration">
-        <li class="active" *ngFor="let iteration of currentIterations" [routerLinkActive]="'selected-iteration'">
+        <li class="active" *ngFor="let iteration of currentIterations" [routerLinkActive]="'selected-iteration'"
```
Comment (sanbornsen) [inline src/app/iteration/iteration.component.html]: So, problem with this code is, you can't really right click on that section and copy the URL ot open it in a new tab. So we need an anchor tag for this.

Comment (sanbornsen) [inline src/app/iteration/iteration.component.html]: So we had it almost like this previously, and that's the reason I changed them to anchor tag. It works fine, so if we think it is okay for the summit, then let's go ahead with this.

Comment (michaelkleinhenz) [inline src/app/iteration/iteration.component.html]: Uhh..We should *not* support open in new tabs anyway at this point. This causes all sorts of problems, lost updates are the least. So I think explicitly disabling that is actually a good thing.

**Comment by michaelkleinhenz** [comment]
Something more general. @sanbornsen the solution with the attributes is really slick and cool. I was first a bit confused and looked how the linking works but as soon as I discovered it is just that two attributes that add routing capabilities to an element I was like "oh, and that's it? that's cool!" :-) Good Work!

**Comment by fabric8cd** [comment]
@michaelkleinhenz snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1766-fabric8-planner.int.rdu2c.fabric8.io

**Comment by fabric8cd** [comment]
@michaelkleinhenz snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1766-fabric8-planner.int.rdu2c.fabric8.io

File: src/app/iteration/iteration.component.html (hunk 1 above)
Comment (sanbornsen) [inline src/app/iteration/iteration.component.html]: If that is the case then all good to go. 👍

**Comment by sanbornsen** [comment]
@michaelkleinhenz thanks. Yeah it's really cool.

**Pull request merged**